src/sinks/honeycomb.rs

```diff
@@ -140,10 +175,21 @@ impl HttpEventEncoder<serde_json::Value> for HoneycombEventEncoder {
             chrono::Utc::now()
         };
 
-        let data = json!({
+        let sample_rate = self
+            .sample_rate_source
+            .field_path()
+            .and_then(|path| log.remove(path))
+            .and_then(|v| v.as_integer());
+
+        let mut data = json!({
             "time": timestamp.to_rfc3339_opts(chrono::SecondsFormat::Nanos, true),
-            "data": log.convert_to_fields(),
+            "data": log.convert_to_fields()
         });
+        if let Some(sample_rate) = sample_rate {
+            data.as_object_mut()
+                .unwrap()
+                .insert("samplerate".to_owned(), sample_rate.into());
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏It's nice to define these raw strings at the top of the file, something like 

 ```Rust
 const HONEYCOMB_SAMPLERATE_FIELD: &str = "samplerate";
 ```